poc-cb-net/cmd/service/cladnet-service.go

```diff
@@ -0,0 +1,167 @@
+
+	// Generate a unique CLADNet ID by the xid package
+	guid := xid.New()
+	CBLogger.Tracef("A unique CLADNet ID: %v", guid)
+	cladnetSpec.Id = guid.String()
+
+	// Currently assign the 1st IP address for Gateway IP (Not used till now)
+	ipv4Address, _, errParseCIDR := net.ParseCIDR(cladnetSpec.Ipv4AddressSpace)
+	if errParseCIDR != nil {
+		CBLogger.Error(errParseCIDR)
+		return nil, status.Errorf(codes.Internal, "Error while parsing CIDR", errParseCIDR)
+	}
+	CBLogger.Tracef("IPv4Address: %v", ipv4Address)
+
+	// Assign gateway IP address
+	// ip := ipv4Address.To4()
+	// gatewayIP := nethelper.IncrementIP(ip, 1)
+	// cladnetSpec.GatewayIP = gatewayIP.String()
+	// CBLogger.Tracef("GatewayIP: %v", cladNetConfInfo.GatewayIP)
+
+	// Put the configuration information of the CLADNet to the etcd
+	bytesCLADNetSpec, _ := json.Marshal(&model.CLADNetSpecification{
+		Id:               cladnetSpec.Id,
+		Name:             cladnetSpec.Name,
+		Ipv4AddressSpace: cladnetSpec.Ipv4AddressSpace,
+		Description:      cladnetSpec.Description})
+	CBLogger.Tracef("CLADNet specification: %v", bytesCLADNetSpec)
+
+	keyConfigurationInformationOfCLADNet := fmt.Sprint(etcdkey.ConfigurationInformation + "/" + cladnetSpec.Id)
+	_, err := etcdClient.Put(context.Background(), keyConfigurationInformationOfCLADNet, string(bytesCLADNetSpec))
+	if err != nil {
+		CBLogger.Error(err)
+		return nil, status.Errorf(codes.Internal, "Error while putting CLADNetSpecification", err)
+	}
+
+	return &pb.CLADNetSpecification{
+		Id:               cladnetSpec.Id,
+		Name:             cladnetSpec.Name,
+		Ipv4AddressSpace: cladnetSpec.Ipv4AddressSpace,
+		Description:      cladnetSpec.Description}, status.New(codes.OK, "").Err()
+}
+
+func (s *server) GetCLADNet(ctx context.Context, in *pb.CLADNetID) (*pb.CLADNetSpecification, error) {
+	log.Printf("Received profile: %v", in)
+	// cladnetResponse := &pb.CLADNetResponse{
+	// 	IsSucceeded: true,
+	// 	Message:     "",
+	// 	CladnetSpecification: &pb.CLADNetSpecification{
+	// 		Id:               "0",
+	// 		Name:             "AAA",
+	// 		Ipv4AddressSpace: "192.168.77.0/28",
+	// 		Description:      "Described"}}
+	var cladnetSpec pb.CLADNetSpecification
+
+	return &cladnetSpec, status.New(codes.OK, "").Err()
+}
+
+func main() {
+	lis, err := net.Listen("tcp", config.GRPC.ListenPort)
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+
+	// etcd Section
+	etcdClient, err = clientv3.New(clientv3.Config{
+		Endpoints:   config.ETCD.Endpoints,
+		DialTimeout: 5 * time.Second,
+	})
+
+	if err != nil {
+		CBLogger.Fatal(err)
+	}
+
+	defer func() {
+		errClose := etcdClient.Close()
+		if errClose != nil {
+			CBLogger.Fatal("Can't close the etcd client", errClose)
+		}
+	}()
+
+	CBLogger.Infoln("The etcdClient is connected.")
+
+	s := grpc.NewServer()
+	pb.RegisterCloudAdaptiveNetworkServer(s, &server{})
+	if err := s.Serve(lis); err != nil {
+		log.Fatalf("failed to serve: %v", err)
+	}
+}
```

Comment:
How about using the full name? `listener`  :)
or .. `ln` ?

```suggestion
	listener, err := net.Listen("tcp", config.GRPC.ListenPort)
	if err != nil {
		log.Fatalf("failed to listen: %v", err)
	}

	// etcd Section
	etcdClient, err = clientv3.New(clientv3.Config{
		Endpoints:   config.ETCD.Endpoints,
		DialTimeout: 5 * time.Second,
	})

	if err != nil {
		CBLogger.Fatal(err)
	}

	defer func() {
		errClose := etcdClient.Close()
		if errClose != nil {
			CBLogger.Fatal("Can't close the etcd client", errClose)
		}
	}()

	CBLogger.Infoln("The etcdClient is connected.")

	s := grpc.NewServer()
	pb.RegisterCloudAdaptiveNetworkServer(s, &server{})
	if err := s.Serve(listener); err != nil {
		log.Fatalf("failed to serve: %v", err)
	}
}
```

Reply:
@seokho-son 

Thank you for your review comments 😍 

I prefer the full name, e.g. listener.

I think it's needed to check again variable names ^^